diff --git a/lib/icinga/checkresult.ti b/lib/icinga/checkresult.ti
index 43c985a..e6eb51b 100644
--- a/lib/icinga/checkresult.ti
+++ b/lib/icinga/checkresult.ti
@@ -73,6 +73,7 @@ class CheckResult
 	[state] String output;
 	[state] Array::Ptr performance_data;
 
+	[state] String perfdata_raw;
 	[state] bool active {
 		default {{{ return true; }}}
 	};
diff --git a/lib/icinga/externalcommandprocessor.cpp b/lib/icinga/externalcommandprocessor.cpp
index e0a6b31..f7bc6e1 100644
--- a/lib/icinga/externalcommandprocessor.cpp
+++ b/lib/icinga/externalcommandprocessor.cpp
@@ -305,6 +305,7 @@ void ExternalCommandProcessor::ProcessHostCheckResult(double time, const std::ve
 	std::pair<String, String> co = PluginUtility::ParseCheckOutput(arguments[2]);
 	result->SetOutput(co.first);
 	result->SetPerformanceData(PluginUtility::SplitPerfdata(co.second));
+	result->SetPerfdataRaw(co.second);
 
 	ServiceState state;
 
diff --git a/lib/icinga/host.cpp b/lib/icinga/host.cpp
index 5ea6696..339d365 100644
--- a/lib/icinga/host.cpp
+++ b/lib/icinga/host.cpp
@@ -295,6 +295,9 @@ bool Host::ResolveMacro(const String& macro, const CheckResult::Ptr&, Value *res
 		} else if (macro == "perfdata") {
 			*result = PluginUtility::FormatPerfdata(cr->GetPerformanceData());
 			return true;
+		} else if (macro == "perfdata_raw") {
+		        *result = cr->GetPerfdataRaw();
+		        return true;
 		} else if (macro == "last_check") {
 			*result = Convert::ToString((long)cr->GetScheduleStart());
 			return true;
diff --git a/lib/icinga/service.cpp b/lib/icinga/service.cpp
index 05db7f3..ffec444 100644
--- a/lib/icinga/service.cpp
+++ b/lib/icinga/service.cpp
@@ -225,6 +225,9 @@ bool Service::ResolveMacro(const String& macro, const CheckResult::Ptr& cr, Valu
 		} else if (macro == "perfdata") {
 			*result = PluginUtility::FormatPerfdata(cr->GetPerformanceData());
 			return true;
+		} else if (macro == "perfdata_raw") {
+		        *result = cr->GetPerfdataRaw();
+		        return true;
 		} else if (macro == "last_check") {
 			*result = Convert::ToString((long)cr->GetExecutionEnd());
 			return true;
diff --git a/lib/methods/pluginchecktask.cpp b/lib/methods/pluginchecktask.cpp
index ec8b882..8816464 100644
--- a/lib/methods/pluginchecktask.cpp
+++ b/lib/methods/pluginchecktask.cpp
@@ -73,6 +73,7 @@ void PluginCheckTask::ProcessFinishedHandler(const Checkable::Ptr& checkable, co
 	cr->SetCommand(commandLine);
 	cr->SetOutput(co.first);
 	cr->SetPerformanceData(PluginUtility::SplitPerfdata(co.second));
+	cr->SetPerfdataRaw(co.second);
 	cr->SetState(PluginUtility::ExitStatusToState(pr.ExitStatus));
 	cr->SetExitStatus(pr.ExitStatus);
 	cr->SetExecutionStart(pr.ExecutionStart);
