diff --git a/library/Icinga/Authentication/AdmissionLoader.php b/library/Icinga/Authentication/AdmissionLoader.php
index 0a80be127..4266bf366 100644
--- a/library/Icinga/Authentication/AdmissionLoader.php
+++ b/library/Icinga/Authentication/AdmissionLoader.php
@@ -40,6 +40,9 @@ class AdmissionLoader
                 }
             }
         }
+	if ($username != "*"){
+	    return $this->match("*",$userGroups,$section);
+	}
         return false;
     }
 
