diff --git a/doc/18-library-reference.md b/doc/18-library-reference.md
index a819bbc..01c74e2 100644
--- a/doc/18-library-reference.md
+++ b/doc/18-library-reference.md
@@ -459,6 +459,23 @@ Example:
     <2> => basename(path)
     "xmpp-notification.pl"
 
+### pathexist <a id="global-functions-pathexist"></a>
+
+Signature:
+
+    function pathexist(path)
+
+Returns true if the path exist otherwise false.
+
+Example:
+
+    $ icinga2 console
+    Icinga 2 (version: v2.7.0)
+    <1> => pathexist("/")
+    true
+    <2> => pathexist("/a/non/exist/path")
+    false
+
 ### escape_shell_arg <a id="global-functions-escape_shell_arg"></a>
 
 Signature:
diff --git a/lib/base/scriptutils.cpp b/lib/base/scriptutils.cpp
index ef70ecc..3195c52 100644
--- a/lib/base/scriptutils.cpp
+++ b/lib/base/scriptutils.cpp
@@ -58,6 +58,7 @@ REGISTER_SAFE_SCRIPTFUNCTION_NS(System, bool, &ScriptUtils::CastBool, "value");
 REGISTER_SAFE_SCRIPTFUNCTION_NS(System, get_time, &Utility::GetTime, "");
 REGISTER_SAFE_SCRIPTFUNCTION_NS(System, basename, &Utility::BaseName, "path");
 REGISTER_SAFE_SCRIPTFUNCTION_NS(System, dirname, &Utility::DirName, "path");
+REGISTER_SAFE_SCRIPTFUNCTION_NS(System, pathexist, &Utility::PathExists, "path");
 REGISTER_SAFE_SCRIPTFUNCTION_NS(System, msi_get_component_path, &ScriptUtils::MsiGetComponentPathShim, "component");
 REGISTER_SAFE_SCRIPTFUNCTION_NS(System, track_parents, &ScriptUtils::TrackParents, "child");
 REGISTER_SAFE_SCRIPTFUNCTION_NS(System, escape_shell_cmd, &Utility::EscapeShellCmd, "cmd");
diff --git a/lib/base/utility.hpp b/lib/base/utility.hpp
index 3e19301..8410b8a 100644
--- a/lib/base/utility.hpp
+++ b/lib/base/utility.hpp
@@ -69,6 +69,7 @@ public:
 
 	static String DirName(const String& path);
 	static String BaseName(const String& path);
+	static bool PathExist(const String& path);
 
 	static void NullDeleter(void *);
 
