diff --git a/lib/icinga/checkresult.ti b/lib/icinga/checkresult.ti
index 1920de5..cbe9b03 100644
--- a/lib/icinga/checkresult.ti
+++ b/lib/icinga/checkresult.ti
@@ -73,6 +73,7 @@ class CheckResult
 	[state] String output;
 	[state] Array::Ptr performance_data;
 
+	[state] String perfdata_raw;
 	[state] bool active {
 		default {{{ return true; }}}
 	};
diff --git a/lib/icinga/externalcommandprocessor.cpp b/lib/icinga/externalcommandprocessor.cpp
index 6c3469f..4ec0262 100644
--- a/lib/icinga/externalcommandprocessor.cpp
+++ b/lib/icinga/externalcommandprocessor.cpp
@@ -324,6 +324,7 @@ void ExternalCommandProcessor::ProcessHostCheckResult(double time, const std::ve
 	std::pair<String, String> co = PluginUtility::ParseCheckOutput(arguments[2]);
 	result->SetOutput(co.first);
 	result->SetPerformanceData(PluginUtility::SplitPerfdata(co.second));
+	result->SetPerfdataRaw(co.second);
 
 	ServiceState state;
 
diff --git a/lib/icinga/host.cpp b/lib/icinga/host.cpp
index c84edda..6c3f9cc 100644
--- a/lib/icinga/host.cpp
+++ b/lib/icinga/host.cpp
@@ -327,6 +327,9 @@ bool Host::ResolveMacro(const String& macro, const CheckResult::Ptr&, Value *res
 		} else if (macro == "perfdata") {
 			*result = PluginUtility::FormatPerfdata(cr->GetPerformanceData());
 			return true;
+		} else if (macro == "perfdata_raw") {
+		        *result = cr->GetPerfdataRaw();
+		        return true;
 		} else if (macro == "check_source") {
 			*result = cr->GetCheckSource();
 			return true;
diff --git a/lib/icinga/service.cpp b/lib/icinga/service.cpp
index 54fa55e..d36606b 100644
--- a/lib/icinga/service.cpp
+++ b/lib/icinga/service.cpp
@@ -258,6 +258,9 @@ bool Service::ResolveMacro(const String& macro, const CheckResult::Ptr& cr, Valu
 		} else if (macro == "perfdata") {
 			*result = PluginUtility::FormatPerfdata(cr->GetPerformanceData());
 			return true;
+		} else if (macro == "perfdata_raw") {
+		        *result = cr->GetPerfdataRaw();
+		        return true;
 		} else if (macro == "check_source") {
 			*result = cr->GetCheckSource();
 			return true;
diff --git a/lib/methods/pluginchecktask.cpp b/lib/methods/pluginchecktask.cpp
index abf2b0c..eae68d7 100644
--- a/lib/methods/pluginchecktask.cpp
+++ b/lib/methods/pluginchecktask.cpp
@@ -77,6 +77,7 @@ void PluginCheckTask::ProcessFinishedHandler(const Checkable::Ptr& checkable, co
 	cr->SetCommand(commandLine);
 	cr->SetOutput(co.first);
 	cr->SetPerformanceData(PluginUtility::SplitPerfdata(co.second));
+	cr->SetPerfdataRaw(co.second);
 	cr->SetState(PluginUtility::ExitStatusToState(pr.ExitStatus));
 	cr->SetExitStatus(pr.ExitStatus);
 	cr->SetExecutionStart(pr.ExecutionStart);
